### PR TITLE
behaviors: don't run behaviors in iframes that are about:blank or are…

### LINKED ISSUE #210: Smarter behavior injection
Currently, behaviors are injected into all frames, mostly for autoplay and autofetch background behaviors.
However, this can result in errors and unnecessary waiting for behaviors in frames that don't need them.
Instead, only inject behavior into main frame and frames that are not ads, using the ad block checking mechanism.

Potentially, also look at CDP [Page.adFrameType](https://chromedevtools.github.io/devtools-protocol/tot/Page/#type-AdFrameType](Page.adFrameType), though currently this may not be working yet.

### PR DESCRIPTION
**Pull request by ikreymer**
… from an ad-host (even if ad-blocking is not disabled)

fixes #210

Possibly could add more logging of whether behaviors are injected or skipped

### CODE REVIEW

File: crawler.js [hunk 1]
```diff
@@ -592,9 +610,9 @@ export class Crawler {
 
     await this.initPages();
 
-    if (this.params.blockAds) {
-      this.adBlockRules = new AdBlockRules(this.captureBasePrefix, this.params.adBlockMessage, this.logger);
-    }
+    //if (this.params.blockAds) {
```
Comment (tw4l) [inline crawler.js:613]: Should we just remove this conditional instead of commenting it out?

File: crawler.js [hunk 2]
```diff
@@ -433,6 +437,20 @@ export class Crawler {
...
+      return true;
+    }
+
+    const url = frame.url();
+
+    if (url === "about:blank") {
+      return false;
+    }
+
+    return await this.adBlockRules.shouldBlock(null, url);
```
Comment (tw4l) [inline crawler.js:451]: I think this will return `null` or `"advertisement"` rather than a boolean - does that work as expected in the filter?

**Review by tw4l — COMMENTED** [review]
A few small questions here but overall looks good!

File: crawler.js (hunk 1 above)
Comment (ikreymer) [reply crawler.js:613]: Yes

File: crawler.js (hunk 2 above)
Comment (ikreymer) [inline crawler.js:451]:
```suggestion
    return !!(await this.adBlockRules.shouldBlock(null, url));
```

Comment (ikreymer) [reply crawler.js:451]: Should just make it a bool to be safe

**Pull request merged**